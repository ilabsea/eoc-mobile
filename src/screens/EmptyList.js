import React from 'react';
import {StyleSheet, ActivityIndicator, View, Text} from 'react-native';
import {H1} from 'native-base';
import {appVersion} from '../config/utils';

const Loading = ({isFetching}) => {
  const opacity = isFetching ? 1.0 : 0.0;
  return (
    <View style={styles.container}>
      <View testID="loading" style={styles.loadingDirection}>
        <ActivityIndicator
          style={{opacity}}
          size={60}
          color="#0000ff"
          animating={true}
        />
      </View>
    </View>
  );
};

const EmptyList = ({isFetching, data}) => {
  return isFetching ? (
    <Loading isFetching={isFetching} />
  ) : data.length === 0 ? (
    <View style={styles.noItem}>
      <H1 testID="emptyMsg">No related items</H1>
      <Text>App version: {appVersion}</Text>
    </View>
  ) : null;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    position: 'absolute',
    alignItems: 'center',
    left: 0,
    right: 0,
    top: 0,
    bottom: 0,
    zIndex: 1,
  },
  noItem: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  loadingDirection: {
    flexDirection: 'row',
  },
});

export default EmptyList;
