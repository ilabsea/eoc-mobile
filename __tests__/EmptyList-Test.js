import React from 'react'
import renderer from 'react-test-renderer'
import EmptyList from '../src/screens/EmptyList'

it('render correctly', () => {
  const tree = renderer.create(<EmptyList isFetching={false} data={[]} />).toJSON()
  expect(tree).toMatchSnapshot()
})