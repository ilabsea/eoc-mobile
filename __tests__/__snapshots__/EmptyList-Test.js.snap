// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 27,
        "lineHeight": 32,
      }
    }
  >
    No related items
  </Text>
  <Text>
    App version: 
  </Text>
</View>
`;
